app/controllers/contacts_controller.rb

```diff
@@ -17,8 +23,16 @@ def create
   	end 
   end 
 
+  def most_recent_appointment(each_contact)
```

Comment:
this method feels like it belongs in the `contact` model rather than in the controller. In Rails, you've probably heard people say "skinny controllers, fat models" -- ideally models aren't too fat, but I think the idea of "keep logic out of controllers as much as possible" is valid